api/v1beta1/nutanixfailuredomain_types.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2025 Nutanix
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+const (
+	// NutanixFailureDomainV2Kind represents the Kind of NutanixFailureDomainV2
+	NutanixFailureDomainV2Kind = "NutanixFailureDomainV2"
+
+	// NutanixFailureDomainV2Finalizer is the finalizer used by the NutanixFailureDomainV2 controller to block
+	// deletion of the NutanixFailureDomainV2 object if there are references to this object by other resources.
+	NutanixFailureDomainV2Finalizer = "nutanixfailuredomainv2.infrastructure.cluster.x-k8s.io"
+)
+
+// PrismElementFailureDomainSpec defines Prism Element specific failure domain spec.
+type PrismElementFailureDomainSpec struct {
+	// prismElement is to identify the Prism Element in the Prism Central for the failure domain.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="prismElement is immutable once set"
+	// +kubebuilder:validation:Required
+	PrismElement NutanixResourceIdentifier `json:"prismElement"`
+
+	// subnets holds a list of identifiers (one or more) of the PE cluster's network subnets
+	// for the Machine's VM to connect to.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="subnets is immutable once set"
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
+	Subnets []NutanixResourceIdentifier `json:"subnets"`
+}
+
+// NutanixFailureDomainV2Spec defines the desired state of NutanixFailureDomainV2.
+type NutanixFailureDomainV2Spec struct {
+	// prismElement defines Prism Element specific failure domain configuration.
+	// +kubebuilder:validation:Required
+	PrismElement PrismElementFailureDomainSpec `json:"prismElement"`
+}
+
+// NutanixFailureDomainV2Status defines the observed state of NutanixFailureDomainV2 resource.
+type NutanixFailureDomainV2Status struct {
+	// conditions represent the latest states of the failure domain.
+	// +optional
+	Conditions []capiv1.Condition `json:"conditions,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=nutanixfailuredomainv2s,shortName=nfd,scope=Namespaced,categories=cluster-api
+// +kubebuilder:subresource:status
+// +kubebuilder:storageversion
+
+// NutanixFailureDomainV2 is the Schema for the NutanixFailureDomainV2 API.
+type NutanixFailureDomainV2 struct {
```

Comment:
Rename the CRD from NutanixFailureDomainV2 to NutanixFailureDomain. API versioning belongs in the group/version (e.g. infrastructure.nutanix/v1beta1), not in the Kind name itself. That keeps our API consistent with Kubernetes best practices and avoids confusion when we introduce future versions (v1beta2, etc.).

Reply:
Renamed to NutanixFailureDomain